aws-api/src/main/java/com/amplifyframework/api/aws/AppSyncGraphQLRequest.java

```diff
@@ -0,0 +1,295 @@
+            List<String> inputKeys = new ArrayList<>(variableTypes.keySet());
+            Collections.sort(inputKeys);
+
+            List<String> inputTypes = new ArrayList<>();
+            List<String> inputParameters = new ArrayList<>();
+            for (String key : inputKeys) {
+                inputTypes.add("$" + key + ": " + variableTypes.get(key));
+                inputParameters.add(key + ": $" + key);
+            }
+
+            inputTypeString = Wrap.inParentheses(TextUtils.join(", ", inputTypes));
+            inputParameterString = Wrap.inParentheses(TextUtils.join(", ", inputParameters));
+        }
+
+        String modelName = modelSchema.getName();
+        String operationString = new StringBuilder()
+                .append(Casing.from(Casing.CaseType.SCREAMING_SNAKE_CASE).to(Casing.CaseType.CAMEL_CASE)
+                        .convert(operationType.toString()))
+                .append(modelName)
+                .append(QueryType.LIST.equals(operationType) ? "s" : "")
+                .append(inputParameterString)
+                .append(selectionSet.toString())
+                .toString();
+
+        String queryString = new StringBuilder()
+                .append(operationType.getOperationName())
+                .append(" ")
+                .append(Casing.from(Casing.CaseType.SCREAMING_SNAKE_CASE).to(Casing.CaseType.PASCAL_CASE)
+                        .convert(operationType.toString()))
+                .append(modelName)
+                .append(inputTypeString)
+                .append(" ")
+                .append(Wrap.inBraces(operationString))
+                .toString();
+
+        return queryString;
+    }
+
+    @Override
+    public boolean equals(Object object) {
+        if (this == object) {
+            return true;
+        }
+
+        if (object == null || getClass() != object.getClass()) {
+            return false;
+        }
+        if (!super.equals(object)) {
+            return false;
+        }
+        AppSyncGraphQLRequest<?> that = (AppSyncGraphQLRequest<?>) object;
+        return ObjectsCompat.equals(modelSchema, that.modelSchema) &&
+                ObjectsCompat.equals(operationType, that.operationType) &&
+                ObjectsCompat.equals(selectionSet, that.selectionSet) &&
+                ObjectsCompat.equals(variables, that.variables) &&
+                ObjectsCompat.equals(variableTypes, that.variableTypes);
+    }
+
+    @Override
+    public int hashCode() {
+        return ObjectsCompat.hash(super.hashCode(), modelSchema, operationType, selectionSet, variables, variableTypes);
+    }
+
+    @Override
+    public String toString() {
+        return "AppSyncGraphQLRequest{" +
+                "modelSchema=" + modelSchema +
+                ", operationType=" + operationType +
+                ", selectionSet=" + selectionSet +
+                ", variables=" + variables +
+                ", variableTypes=" + variableTypes +
+                super.toString() +
```

Comment:
Hm, is this gonna work? It'll output something like `GraphQLRequest { ... }` in the middle. Would it be better to just access the `super.field`s directly to flatten them inline with these others?

Reply:
It would work in the sense that it would print out all the properties, but I agree, with a weird nested `GraphQLRequest { }`.  I just updated it just print them all inline with the others.